server/src/main/java/org/elasticsearch/common/time/DateFormatter.java

```diff
@@ -30,6 +31,14 @@
      */
     TemporalAccessor parse(String input);
 
+    /**
+     * Try to parse input to a java time TemporalAccessor
+     * @param input An arbitrary string resembling the string representation of a date or time
+     * @return      Tuple containing a boolean value indicating parsing success or failure and a java time object containing
+     * the parsed input in the case of successful parsing
+     */
+    Tuple<Boolean, TemporalAccessor> parseWithoutException(String input);
```

Comment:
Is it expected that a success could result in a null `TemporalAccessor`? That is, is it acceptable to return `Tuple.tuple(true, null)`?

If not, what about `Optional<TemporalAccessor>`?

Reply:
Good idea. Done in https://github.com/elastic/elasticsearch/pull/83801/commits/15622ec612a3c30324754758328f9b4d23cd51ee